coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
After examining further, it appears that the issue for me is, ```this``` is not bound to the step as it should be. You should get errors on ```this.trigger('before-show')``` and such, because it expects this to be the step.

I'm 100% definitely seeing this behavior, so we need to bind ```this``` somehow. Is having one extra arrow detrimental to the code?